### PR TITLE
return skip on channel close

### PR DESCRIPTION
**Pull request by tkumor3**
Closes #

## Introduced changes

-
-

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Review by cptartur — APPROVED** [review]

File: crates/forge/src/test_case_summary.rs [hunk 1]
```diff
@@ -44,10 +44,10 @@ pub enum TestCaseSummary {
         /// Name of the test case
         name: String,
     },
-    /// Fuzzing subtests skipped (did not run), previous subtest failed
-    SkippedFuzzing {},
-    /// Test case execution interrupted by error (did not run or was cancelled)
-    InterruptedByError {},
+    /// Fuzzing subtests skipped (did not run), previous subtest failed or
+    /// Test case execution interrupted by error (did not run or was cancelled).
+    /// To summing up this enum is return when we want to ignore test result
```
Comment (piotmag769) [inline crates/forge/src/test_case_summary.rs]: I would improve this description, it is not clear I think

**Review by piotmag769 — APPROVED** [review]

File: crates/forge/src/lib.rs [hunk 2]
```diff
@@ -334,13 +334,18 @@ async fn run_tests_from_crate(
...
 
     let mut results = vec![];
+    let mut interrupted = false;
 
     while let Some(task) = tasks.next().await {
         let result = task??;
+        match result {
+            TestCaseSummary::Interrupted {} => interrupted = true,
+            result => {
+                pretty_printing::print_test_result(&result);
```
Comment (piotmag769) [inline crates/forge/src/lib.rs:344]: Good job! Can we add a comment explaining why it is done this way though?

File: crates/forge/src/lib.rs [hunk 3]
```diff
@@ -349,6 +354,8 @@ async fn run_tests_from_crate(
     drop(send_shut_down);
     let _ = rec_shut_down.recv().await;
 
+    assert!(!interrupted, "Tests were interrupted");
```
Comment (cptartur) [inline crates/forge/src/lib.rs:357]: I think a comment explaining the logic behind this assert could be useful here. It's not 100% obvious why we want to panic if this is reached.

File: crates/forge/src/lib.rs [hunk 4]
```diff
@@ -349,6 +356,11 @@ async fn run_tests_from_crate(
     drop(send_shut_down);
     let _ = rec_shut_down.recv().await;
 
+    // This Panic should never occur.
+    // If TestCaseSummary::Interrupted is returned by a test,
+    // this implies that there's another return with an Err result.
```
Comment (cptartur) [inline crates/forge/src/lib.rs:361]:
```suggestion
    // If TestCaseSummary::Interrupted is returned by a test,
    // this implies that there should be another test than returned an Err.
```

File: crates/forge/src/test_case_summary.rs [hunk 5]
```diff
@@ -44,10 +44,11 @@ pub enum TestCaseSummary {
...
         name: String,
     },
-    /// Fuzzing subtests skipped (did not run), previous subtest failed
-    SkippedFuzzing {},
-    /// Test case execution interrupted by error (did not run or was cancelled)
-    InterruptedByError {},
+    /// Enum variant representing:
+    ///  - fuzzing subtest that was skipped/interrupted during fuzzing due to other subtest failing
+    ///  - single test or fuzzing subtest that was interrupted by error
+    /// This enum is returned when we want to ignore the test result
```
Comment (cptartur) [inline crates/forge/src/test_case_summary.rs:50]:
```suggestion
    /// Test case execution interrupted:
    ///
    /// Possible causes:
    ///  - fuzzing subtest that was skipped/interrupted during fuzzing due to other subtest failing
    ///  - single test or fuzzing subtest that was interrupted by error
    /// This enum is returned when we want to ignore the test result
```

**Review by cptartur — APPROVED** [review]

**Pull request merged**